server/stream.go

```diff
@@ -3486,13 +3488,15 @@ func (mset *stream) storeMsgId(dde *ddentry) {
 // storeMsgIdLocked will store the message id for duplicate detection.
 // Lock should he held.
 func (mset *stream) storeMsgIdLocked(dde *ddentry) {
-	if mset.ddmap == nil {
-		mset.ddmap = make(map[string]*ddentry)
-	}
-	mset.ddmap[dde.id] = dde
-	mset.ddarr = append(mset.ddarr, dde)
-	if mset.ddtmr == nil {
-		mset.ddtmr = time.AfterFunc(mset.cfg.Duplicates, mset.purgeMsgIds)
+	if mset.cfg.Duplicates != time.Duration(0) {
```

Comment:
Same here `> 0`

Reply:
same as above, changed to `> time.Duration(0)`